app/models/foreman_tasks/task.rb

```diff
@@ -223,6 +227,13 @@ def to_label
       parts.join(' ').strip
     end
 
+    def output_by_type
+      return {} unless main_action && main_action.live_output.any?
+      main_action.live_output.each_with_object({}) do |output, o_hash|
```

Comment:
would it be more readable if we hardcode 'stderr' and 'stdout' here? I don't think we'll see any more output tymes in foreseeable future :-)

also I think it should always return hash with stdout and stderr keys, they should just have empty value

something like
```
return { :stderr => nil, :stdout => nil} unless main_action && main_action.live_output.any?
{ :stderr => ..., :stdout => ... }
```

Reply:
I like to avoid hardcoding because it tends to create piece of code that is forgotten when the new "out" (ie. `stdwarn`) stream is created. I know it's a extreme example but I don't know how frequently can happen. What do you think  @ares

Reply:
There are more places where we hardcode this, e.g. https://github.com/theforeman/foreman_remote_execution/blob/master/app/views/template_invocations/show.html.erb#L17-L22

if user defines STDWARN, our tool would somehow have to detecte it, but the way we run SSH / Ansible does not allow it today. Given the chance here is close to zero, I'd be fine with hardcoding.

But if you feel strongly it's better to be smart here and iterate over all available statuses, I don't insist :-) I'd find the return value of this method more readable, perhaps adding a comment describing the result would do.